src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
```

Comment:
> note that your queried maxPath value can be -1

It actually won't be here, due to https://github.com/dotnet/corefx/pull/2248/files#diff-57471f013b586ed190c3bbce817edd1cR39